### DISCUSSION TITLE
Enhance song request limit parameters (allowing X requests within X minutes + flood protection)

### DISCUSSION THREAD

**Discussion by RM-FM** [AzuraCast/AzuraCast#6937] [Ideas]
Currently the only option to limit song requests is 1 song within X minutes per user/IP. It would be great to have more flexibility by adding a "X songs within X minutes" option. Additionally, a station wide max. limit could (or should!) act as a flood protection, independently from the IP. The latter could be bound to the number of pending requests.

**Mock-up "Edit Station Profile > Song Requests"**
![image](https://github.com/AzuraCast/AzuraCast/assets/40538914/f12c812f-cf90-44c5-9acf-bad6257d3f24)

**Comment by sdensler** [comment]
Love this idea...we have some superfans that request the same artist over and over and we don't want to dissuade them from participating but it has gotten to be too much and very noticeable to other listeners.
